hashicorp/go-tfe
# add agent data source

## Description

This PR adds a new data source to the Client. The purpose is to allow users to Read, List agents. Please see the Agent Pool and Agents documentation, linked below, for more information. 

Please be aware that this PR downloads the [tfc-agent binary](https://releases.hashicorp.com/tfc-agent/), which only supports Linux AMD64 architecture.

## Testing plan

1. Checkout this branch
2. Ensure you have an existing agent pool (Org -> Settings -> Agent Pool)
3. Ensure you have an agent pool token
4. Create an agent
5. Make an API call to read, list an agent

-- or --

1. With the required architecture
2. Run the test command noted, below
3. The test suite should pass ✅ 

## External links

- [Agent Pools and Agents API documentation](https://www.terraform.io/cloud-docs/api-docs/agents#agent-pools-and-agents-api)

## Output from tests
PASS
ok  	github.com/hashicorp/go-tfe

See example of how to run integration tests, below. Refer to [TESTS.md](https://github.com/hashicorp/go-tfe/blob/main/docs/TESTS.md) for further details.

```
envchain tfe.local go test ./... -v -tags=integration -run TestAgents

...
```


---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202267776134371
